packages/accordion/src/useAccordion.ts

```diff
@@ -5,49 +5,84 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { useState } from 'react';
+import { useState, HTMLProps } from 'react';
 import {
   composeEventHandlers,
   generateId,
   KEY_CODES,
   getControlledValue
 } from '@zendeskgarden/container-utilities';
 
+export interface IUseAccordionProps {
+  idPrefix?: string;
+  expandedSections?: number[];
+  onChange?: (expanded: number[]) => any;
+  expandable?: boolean;
+  collapsible?: boolean;
+}
+
+interface IHeaderProps extends HTMLProps<any> {
+  ariaLevel?: number | null;
+  role?: any;
+}
+
+interface ITriggerProps extends HTMLProps<any> {
+  index?: number;
+  role?: any;
+  tabIndex?: any;
```

Comment:
Is `tabIndex` a valid member of `HTMLProps<any>`? You may not need it here.

Reply:
Same with `role` as well.

Reply:
I believe `tabIndex` and `role` are valid members, however, [our current API uses `null` as a potential value](https://github.com/zendeskgarden/react-containers/blob/master/packages/accordion/stories.js#L50), and that isn't allowed. I can remove both `role` and `tabIndex` here, but TypeScript will complain when consumers pass in `null` as the value for either of these properties.

Is there a way I can remove `role` and `tabIndex` here, and have TypeScript be happy with using `null` as a value for these properties?

Reply:
Ah, that makes sense. Could it be that we _shouldn't_ be passing in `null` as a value and use `undefined` instead? These seems more like a valid failure than a type miss-match.

Reply:
> Could it be that we shouldn't be passing in null as a value and use undefined instead? These seems more like a valid failure than a type miss-match.

I felt the same way, but I was cautious and I didn't want to come in guns blazing changing existing APIs. I'm glad you pointed this out!

Reply:
For many of our aria attributes, `undefined` is considered an error because the consumer forgot to correctly apply them. We use `null` as an indication that the consumer knows what they are doing (i.e. "I've selected the appropriate semantic HTML tag and these roles/attributes should not be applied"). I don't think we want to change this to rely on `undefined`.

Reply:
Depending on how things are spread onto a component `undefined` may not actually render `role="undefined"`, it could just be ignored by React. @hzhu and I will take a deeper look.

Reply:
But the API would be wrong. If `ariaLevel` is undefined, it's a problem. If it's null, then I'm indicating that I'm using the appropriate `<h3>` (or whatever) tag. And we wouldn't want `role` to behave differently.

Reply:
Ohhhhh yes. That's correct. This may be something we want to have someone force-type in the example or add nullable `| null` types to the values.

Reply:
Great, thanks for your thoughts. I'm keeping it the way it is per the discussion above and after talking it out with @austin94.